apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-common/src/main/java/co/elastic/apm/agent/hibernate/search/HibernateSearchHelper.java

```diff
@@ -0,0 +1,64 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.hibernate.search;
+
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+
+@VisibleForAdvice
+public final class HibernateSearchHelper {
+
+    private HibernateSearchHelper() {
+
+    }
+
+    @VisibleForAdvice
+    public static Span createAndActivateSpan(final ElasticApmTracer tracer, final String methodName,
+                                             final String query) {
+        Span span = null;
+
+        if (tracer != null) {
+            TraceContextHolder<?> active = tracer.getActive();
+            if (active == null || active instanceof Span && HibernateSearchConstants.HIBERNATE_SEARCH_ORM_TYPE
+                .equals(((Span) active).getSubtype())) {
+                return null;
+            }
+
+            span = active.createSpan().activate();
+
+            span.withType("db")
```

Comment:
We should discuss using more descriptive types here.

For the upcoming breakdown graphs it would use `db.hibernate` as the key which does not seem to informative. I'm not sure if `db` is the best type for this, as the query is not done on the `db` but rather with lucene, right? But we also don't want to be too specific on the underlying technology as there are other options to lucene. Is that correct?

Maybe something like `search.hibernate.list`. The action would be the method signature like `list`, `scroll` or `iterate`.

Reply:
Yeah, this is basically one of the hardest things about this PR 😉

1. Query is done in Lucene/Elasticsearch **but** the result is fetched from the database. So it's a hybrid approach.
2. Yes, underlying technology can differ. Currently Lucene or Elasticsearch.

Do you mean `search.hibernate.list` as the whole type? I don't know if putting the methodname there is a good idea for grouping the results.

Reply:
No, I meant `search` as the type, `hibernate` as the subtype and `list` as the action. @eyalkoren WDYT?

Reply:
> Query is done in Lucene/Elasticsearch but the result is fetched from the database. So it's a hybrid approach.

Ah, I see. So maybe `type=db` `subtype=hibernate-search` and `action=list`?

Reply:
I think this is good. Should the db type in the context then also changed to `hibernate-search`?

Reply:
I guess it doesn't make sense to have the type and subtype set to the same value. But maybe `type=hibernate-search` `subtype=lucene/elasticsearch` and `action=list/scroll/iterate`? But I think I like `type=db` `subtype=hibernate-search` and `action=list/scroll/iterate` more.

Reply:
I didn't meant the type of the span with my comment but the type set in line 58 for the context in the span.

Reply:
Ah, gotcha. I think for the `db.type` `hibernate-serach` makes sense. Question: would there typically be a child span under a `hibernate-search` span for the JDBC query which is sent to the DB?

Reply:
There are currently no spans on purpose. See [this comment](https://github.com/elastic/apm-agent-java/pull/685#issuecomment-506725801).

Reply:
I adjusted the types to your suggestion